### PR TITLE
feat: create GrafanaDashboard CR for GrafanaOperator

### LINKED ISSUE #220: [Feature]: create Grafana-Dashboard CR optionally
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've discussed this feature request in the [K8sGPT Slack](https://k8sgpt.slack.com) and got positive feedback

### Is this feature request related to a problem?

No

### Problem Description

This issue is very similar to https://github.com/kyverno/kyverno/pull/7992 . When using the grafana-operator, the grafana-dashboards should be defined in special Grafana-Dashboard CRs and not just a ConfigMap.
So it would be great if the Helm-Charts creates also a Grafana-Dashboard CR and referenced to the ConfigMap (opt-in).

### Solution Description

very similar to https://github.com/kyverno/kyverno/pull/7992

### Benefits

User who have the grafana-operator deployt get automatically the K8sGPT Grafana Dashboard.

### Potential Drawbacks

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by jkleinlercher**
Closes #220

## 📑 Description
User who have the grafana-operator deployed get automatically the K8sGPT Grafana Dashboard.

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: chart/operator/templates/grafana-k8sgpt-dashboard.yaml [hunk 1]
```diff
@@ -15,4 +15,25 @@ metadata:
...
+metadata:
+  labels:
+    app.kubernetes.io/component: grafana-dashboard
+    app.kubernetes.io/created-by: k8sgpt-operator
+    app.kubernetes.io/part-of: k8sgpt-operator
+  {{- include "chart.labels" . | nindent 4 }}
+  name: k8sgpt-overview
+  namespace: {{ .Values.grafanaDashboard.namespace | default (include "k8sgpt-operator.namespace" .) }}
+spec:
+  folder: {{ .Values.grafanaDashboard.folder.name }}
```
Comment (AlexsJones) [inline chart/operator/templates/grafana-k8sgpt-dashboard.yaml:31]: .Values.grafanaDashboard.folder.name does not exist, shall we placeholder?

Comment (jkleinlercher) [reply chart/operator/templates/grafana-k8sgpt-dashboard.yaml:31]: Hm I actually just reused it from https://github.com/k8sgpt-ai/k8sgpt-operator/blob/a5a0283f6239787c5264e388b3347001c6a93401/chart/operator/templates/grafana-k8sgpt-dashboard.yaml#L6

and it is defined in https://github.com/k8sgpt-ai/k8sgpt-operator/blob/0a32492cc1b3b7d7211169f40652b09feab5ff84/chart/operator/values.yaml#L12

Comment (AlexsJones) [reply chart/operator/templates/grafana-k8sgpt-dashboard.yaml:31]: I presume this is due to the prometheus operator having grafana as a sub chart

Comment (jkleinlercher) [reply chart/operator/templates/grafana-k8sgpt-dashboard.yaml:31]: I am afraid I don't understand the discussion. Maybe you can help me with that here ... I can only say that https://github.com/k8sgpt-ai/k8sgpt-operator/blob/main/chart/operator/values.yaml already defines .Values.grafanaDashboard and that is what I use in https://github.com/k8sgpt-ai/k8sgpt-operator/blob/a5a0283f6239787c5264e388b3347001c6a93401/chart/operator/templates/grafana-k8sgpt-dashboard.yaml .. However, maybe I overlook something you think of. Sorry that I don't see the problem here ...

**Review by AlexsJones — APPROVED** [review]
@k8sgpt-ai/k8sgpt-maintainers please review

**Comment by JuHyung-Son** [comment]
after i installed grafana operator. i confirmed `GrafanaDashboard` is created and k8sgpt-overview cm is correctly injected. i haven't used grafana operator, but it seems works well.

```
┌───────────────────────────────────────────────────────────────────────────────────────────────── Grafanadashboards(all)[1] ──────────────────────────────────────────────────────────────────────────────────────────────────┐
│ NAMESPACE↑                                NAME                                           NO MATCHING INSTANCES                               LAST RESYNC                               AGE                                   │
│ default                                   k8sgpt-overview                                true                                                2m26s↑                                    2m26s                                 │
```

**Pull request merged**